code/perception/src/panoptic_segmentation/efficientps/semantic_head/semantic_head.py

```diff
@@ -0,0 +1,247 @@
+from math import ceil
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+from inplace_abn import InPlaceABN
+
+try:
+    from panoptic_segmentation. \
+        efficientps.utils.depthwise_separable_conv import \
+        DepthwiseSeparableConv
+except ImportError:
+    from efficientps.utils.depthwise_separable_conv import \
+        DepthwiseSeparableConv
+
+
+class SemanticHead(nn.Module):
+    """
+    Semantic Head compose of three main module DPC, LSFE and MC
+    Args:
+    - nb_class (int) : number of classes in the dataset
+    """
+
+    def __init__(self, nb_class):
+        super().__init__()
+
+        self.dpc_x32 = DPC()
+        self.dpc_x16 = DPC()
+
+        self.lsfe_x8 = LSFE()
+        self.lsfe_x4 = LSFE()
+
+        self.mc_16_to_8 = MC()
+        self.mc_8_to_4 = MC()
+
+        self.last_conv = nn.Conv2d(512, nb_class, 1)
+
+        self.softmax = nn.Softmax(dim=1)
+        self.cross_entropy_loss = nn.CrossEntropyLoss(reduction='none')
+
+    def forward(self, inputs, targets={}):
+        # TODO Make a loop
```

Comment:
Here is still a todo. Do you intent to make a loop?

Reply:
Nope, was just a todo added by the creator of the code. Todo is removed now :)